docs/fastdds/xml_configuration/dynamic_types.rst

```diff
@@ -246,63 +417,123 @@ child elements.
 The ``<discriminator>`` must define its ``type``
 Each ``<case>`` element has one or more ``<caseDiscriminator>`` elements, which type must be consistent with the
 ``<discriminator>`` type, and a unique ``<member>`` element.
-Please, refer to :ref:`dynamictypes_supportedtypes_union` for more information on the ``<union>`` type.
+Please, refer to :ref:`xtypes_supportedtypes_union` for more information on the ``<union>`` type.
+
+.. tabs::
+
+    .. tab:: XML
 
-.. literalinclude:: /../code/XMLTester.xml
-  :language: xml
-  :start-after: <!-->XML-UNION<-->
-  :end-before: <!--><-->
+        .. literalinclude:: /../code/XMLTester.xml
+            :language: xml
+            :start-after: <!-->XML_UNION<-->
+            :end-before: <!--><-->
+
+    .. tab:: C++
+
+        .. literalinclude:: /../code/DDSCodeTester.cpp
+            :language: c++
+            :start-after: //!--CPP_UNION
+            :end-before: //!--
+
+For a full example of how to define this type, please refer to :ref:`Unions <xtypes_supportedtypes_union>`.
+
+.. _xmldynamictypes_bitset:
 
 Bitset
 """"""
 
-The ``<bitset>`` element defines the :ref:`dynamictypes_supportedtypes_bitset` type.
+The ``<bitset>`` element defines the :ref:`xtypes_supportedtypes_bitset` type.
 It is comprised by a ``name`` attribute and a set of ``<bitfield>`` child elements.
 In turn, the ``<bitfield>`` element has the mandatory ``bit_bound`` attribute, which cannot be higher than 64, and
 two optional attributes: ``name`` and ``type``.
 A ``<bitfield>`` without ``name`` attribute is an inaccessible set of bits.
 Its management ``type`` can ease the ``<bitfield>`` modification and access.
-Please, refer to :ref:`dynamictypes_supportedtypes_bitset` for more information about the ``<bitset>`` type.
+Please, refer to :ref:`xtypes_supportedtypes_bitset` for more information about the ``<bitset>`` type.
+
+.. tabs::
+
+    .. tab:: XML
 
-.. literalinclude:: /../code/XMLTester.xml
-  :language: xml
-  :start-after: <!-->XML-BITSET<-->
-  :end-before: <!--><-->
+        .. literalinclude:: /../code/XMLTester.xml
+            :language: xml
+            :start-after: <!-->XML_BITSET<-->
+            :end-before: <!--><-->
+
+    .. tab:: C++
+
+        .. literalinclude:: /../code/DDSCodeTester.cpp
+            :language: c++
+            :start-after: //!--CPP_BITSET
+            :end-before: //!--
+
+For a full example of how to define this type, please refer to :ref:`Bitset <xtypes_supportedtypes_bitset>`.
+
+.. _xmlxtypes_bitset_inheritance:
+
+Moreover, bitsets can inherit from another bitsets.
+This is implemented by defining the value of the ``baseType`` attribute, on the child ``<bitset>`` element to be the
+value of the ``name`` attribute of the parent ``<bitset>`` element.
+This is exemplified by the code snippet below.
```

Comment:
Are we sure that XML Parser supports bitset inheritance?

Reply:
Not yet but it should. I have left a TODO in 1f40ec6245de6255b8f3189ef10ca2fdf1c17fea